src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/UriExtensions.cs

```diff
@@ -22,6 +23,27 @@ public static string GetAbsoluteOrUNCPath(this Uri uri)
         }
 
         // Absolute paths are usually encoded.
-        return uri.AbsolutePath.Contains("%") ? WebUtility.UrlDecode(uri.AbsolutePath) : uri.AbsolutePath;
+        var absolutePath = uri.AbsolutePath.Contains("%") ? WebUtility.UrlDecode(uri.AbsolutePath) : uri.AbsolutePath;
+
+        if (uri.Scheme.ToLower() == "git")
+        {
+            // return a normalized path when we want to add to a fake git directory path (hacky fix for https://github.com/dotnet/razor/issues/9365)
+            return AddToFakeGitDirectoryAtRoot(absolutePath);
+        }
+
+        return absolutePath;
+    }
+
+    private static string AddToFakeGitDirectoryAtRoot(string decodedAbsolutePath)
+    {
+        var normalizedPath = FilePathNormalizer.Normalize(decodedAbsolutePath);
+        var firstSeparatorIndex = normalizedPath.IndexOf('/');
+        if (firstSeparatorIndex < 0)
+        {
+            // no-op
+            return decodedAbsolutePath;
+        }
+
+        return normalizedPath[..(firstSeparatorIndex + 1)] + "_git_" + normalizedPath[firstSeparatorIndex..];
```

Comment:
Please use `string.Insert` instead of concatenating the result of multiple substrings. It'd be more efficient to insert `_git_/` at the correct index.